### PR TITLE
UPSTREAM: 22634: kubectl: print errors that wont be reloaded in the editor

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/openshift/origin/issues/7811

@smarterclayton

### CODE REVIEW

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/edit.go [hunk 1]
```diff
@@ -294,14 +295,15 @@ outter:
 			}
 
 			if hold, msg := jsonmerge.TestPreconditionsHold(patch, preconditions); !hold {
-				fmt.Fprintf(out, "error: %s", msg)
+				fmt.Fprintf(out, "Error: %s\n", msg)
```
Comment (smarterclayton) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/edit.go]: "error:"

Comment (kargakis) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/edit.go]: We assume that errors printed by fmt.Errorf or errors.New will be preceded by something else hence the convention to lowercase those. This is starting a new line, and I am also consistent with https://github.com/kubernetes/kubernetes/blob/8b186991e2a85731943e099efcd3c676a6f1c774/pkg/kubectl/cmd/edit.go#L400-L406 which are also going to be printed to out.

Comment (smarterclayton) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/edit.go]: We use `error:` in pkg/kubectl/cmd/util#HandleErrors, so we should use `error:` in edit.

Comment (kargakis) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/edit.go]: ok, updated. Can you also please review the upstream patch at https://github.com/kubernetes/kubernetes/pull/22634

**Comment by kargakis** [comment]
@pweil- fyi

**Comment by kargakis** [comment]
upstream has lgtm, [merge]

**Pull request merged**